File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/BikeApi.java
package com.lattis.ellipse.data.network.api;

import com.lattis.ellipse.data.network.model.body.bike.BikeDetailBody;
import com.lattis.ellipse.data.network.model.body.bike.BikeMetaDataBody;
import com.lattis.ellipse.data.network.model.body.bike.BookBikeBody;
import com.lattis.ellipse.data.network.model.body.bike.CancelBikeBody;
import com.lattis.ellipse.data.network.model.body.bike.FindBikesBody;
import com.lattis.ellipse.data.network.model.response.bike.BikeDetailResponse;
import com.lattis.ellipse.data.network.model.response.bike.FindBikeResponse;
import com.lattis.ellipse.data.network.model.response.bike.ReserveBikeResponse;

import retrofit2.http.Body;
import retrofit2.http.POST;
import io.reactivex.Observable;

public interface BikeApi {
    // @POST("bikes/find-bikes/")
    @POST("bikes/search-bikes/")
    Observable<FindBikeResponse> findBikes(@Body FindBikesBody findBikesBody);

    @POST("bikes/create-booking/")
    Observable<ReserveBikeResponse> bookBikes(@Body BookBikeBody bookBikeBody);

    @POST("bikes/cancel-booking/")
    Observable<Void> cancelBikes(@Body CancelBikeBody bookBikeBody);


    @POST("bikes/get-bike-details/")
    Observable<BikeDetailResponse> getBikeDetails(@Body BikeDetailBody bikeDetailBody);

    @POST("bikes/update-metadata-for-user/")
    Observable<Void> updateBikeMetaData(@Body BikeMetaDataBody bikeMetaDataBoday);
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/bike/BikeApi.kt
package com.lattis.data.repository.datasources.api.bike

import com.lattis.data.entity.body.bike.*
import com.lattis.data.entity.response.BasicResponse
import com.lattis.data.entity.response.bike.*
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.*

interface BikeApi{

    @GET("api/rentals")
    fun searchBike(@Query("ne")latLongNE:String, @Query("sw")latLongSW:String): Observable<BikeSearchResponse>

    @GET("api/rentals/find")
    fun findByQRCode(@Query("qr_code")qr_code:String): Observable<FindByQRCodeResponse>

    @POST("api/bikes/create-booking/")
    fun bookBike(@Body bookBikeBody: BookBikeBody): Observable<ReserveBikeResponse>

    @POST("api/bikes/get-bike-details/")
    fun bikeDetails(@Body bikeDetailBody: BikeDetailBody): Observable<BikeDetailResponse>

    @POST("api/bikes/update-metadata-for-user/")
    fun updateBikeMetaData(@Body bikeMetaDataBoday: BikeMetaDataBody): Observable<BasicResponse>

    @POST("api/bikes/cancel-booking/")
    fun cancelBikes(@Body bookBikeBody: CancelBikeBody): Observable<CancelBikeReservationResponse>

    @POST("/api/bikes/{bikeId}/user-lock")
    fun lockIotBike(@Path("bikeId") bike_id:Int, @Body controllerKeyBody: ControllerKeyBody):Observable<IoTBikeLockUnlockCommandStatusResponse>

    @POST("/api/bikes/{bikeId}/user-unlock")
    fun unLockIotBike(@Path("bikeId") bike_id:Int, @Body controllerKeyBody: ControllerKeyBody):Observable<IoTBikeLockUnlockCommandStatusResponse>

    @GET("/api/bikes/{bikeId}/iot/status")
    fun getIotBikeStatus(@Path("bikeId") bike_id:Int,@Query("controller_key")controller_key:String?):Observable<IoTBikeStatusResponse>

    @GET("/api/bikes/{bikeId}/command/{commandId}")
    fun getLinkaIotBikeStatus(@Path("bikeId") bike_id:Int,@Path("commandId") commandId:String):Observable<GetIoTBikeLockUnlockStatusResponse>

    @GET("api/rentals/search")
    fun searchBikeByName(@Query("bike_name")bike_name:String?):Observable<SearchBikeByNameResponse>

    @POST("api/sentinel/{bikeId}/unlock")
    fun unlockSentinel(@Path("bikeId") bike_id:Int):Observable<BasicResponse>



}